protocol/test/forge/templegold/TempleGoldStaking.t.sol

```diff
@@ -423,6 +423,121 @@ contract TempleGoldStakingTest is TempleGoldStakingTestBase {
+        // > is reached for a staker... Thus the wait to get to half of your st-yETH voting power is X days.
+        //
+        // So if we set this to 8 weeks, it should have half the voting power at 8 weeks
+        _setHalftime(8 weeks);
+
+        // Set right on the week boundry to get exact 
+        vm.warp(block.timestamp / WEEK_LENGTH * WEEK_LENGTH);
+
+        // Alice stakes 100
+        vm.startPrank(alice);
+        deal(address(templeToken), alice, 100 ether, true);
+        _approve(address(templeToken), address(staking), type(uint).max);
+        staking.stake(100 ether);
+
+        // Immediately, Alice's voting power is... 100?
+        // @todo @princetonbishop This looks wrong - shouldn't have a 100% voting power
+        assertEq(staking.getVoteWeight(alice), 99.718409010911650827e18);
+
+        // At the end of the first week
+        skip(1 weeks);
+        assertEq(staking.getVoteWeight(alice), 11.111111111111111111e18);
+
+        // At the exact half time, it should be about half the voting power
+        skip(7 weeks);
+        assertEq(staking.getVoteWeight(alice), 50e18);
+
+        // After another 8 weeks it's another half-life amount
+        skip(8 weeks);
+        assertEq(staking.getVoteWeight(alice), 66.666666666666666666e18);
+
+        // And again
+        skip(8 weeks);
+        assertEq(staking.getVoteWeight(alice), 75e18);
+    }
+
+    function test_multiDelegate() public {
+        _setHalftime(8 weeks);
+
+        // Set right on the week boundry to get exact 
+        vm.warp(block.timestamp / WEEK_LENGTH * WEEK_LENGTH);
+
+        vm.startPrank(executor);
+        staking.setVoteDelegate(mike, true);
+
+        // Alice stakes 100 and delegates to Mike
+        {
+            vm.startPrank(alice);
+            deal(address(templeToken), alice, 100 ether, true);
+            _approve(address(templeToken), address(staking), type(uint).max);
+            staking.stake(100 ether);
+            staking.setUserVoteDelegate(mike);
+        }
+
+        // Bob stakes 100 and delegates to Mike
+        {
+            vm.startPrank(bob);
+            deal(address(templeToken), bob, 100 ether, true);
+            _approve(address(templeToken), address(staking), type(uint).max);
+            staking.stake(100 ether);
+            staking.setUserVoteDelegate(mike);
+        }
+
+        // Mike's delegated weight = alice's weight + bob's weight
+        // @todo @princetonbishop This looks wrong - shouldn't have a 100% voting power
+        _checkDelegatedWeight(99.718409010911650827e18, 99.718409010911650827e18);
+
+        skip(1 weeks);
+        _checkDelegatedWeight(11.111111111111111111e18, 11.111111111111111111e18);
+
+        skip(7 weeks);
+        _checkDelegatedWeight(50e18, 50e18);
+
+        skip(16 weeks);
+        _checkDelegatedWeight(75e18, 75e18);
+
+        // Bob stakes another 50 (so his voting power increases by 50%)
```

Comment:
from my local tests. Bob's vote power doesn't increase immediately after he stakes 50e18 more. I think that's the intension for the halfTime to apply to the additional stake